src/mlpack/methods/ann/layer/softshrink.hpp

```diff
@@ -0,0 +1,123 @@
+/**
+ * @file softshrink.hpp
+ * @author Lakshya Ojha
+ *
+ * The soft shrink function has threshold proportional to the noise level given
+ * by the user.
+ * The use of a Soft Shrink activation function provides adaptive denoising at
+ * various noise levels using a single CNN(Convolution Neural) without a
+ * requirement to train a unique CNN for each noise level.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_SOFTSHRINK_HPP
+#define MLPACK_METHODS_ANN_LAYER_SOFTSHRINK_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artifical Neural Network. */ {
+
+/**
+ * Soft Shrink operator is defined as,
+ * @f{eqnarray*}{
+ * f(x) &=& \left\{
+ *   \begin{array}{lr}
+ *     x - lambda & : x >  lambda \\
+ *     x + lambda & : x < -lambda \\
+ *     0 & : otherwise
+ *   \end{array} \\
+ * \right.
+ * f'(x) &=& \left\{
+ *   \begin{array}{lr}
+ *     1 & : x >  lambda \\
+ *     1 & : x < -lambda \\
+ *     0 & : otherwise
+ *   \end{array}
+ * \right.
+ * @f}
+ * lambda is set to 0.5 by default.
+ */
+template <
+    typename InputDataType = arma::mat,
+    typename OutputDataType = arma::mat
+>
+class SoftShrink
+{
+ public:
+  /**
+   * Create Soft Shrink object using specified hyperparameter lambda.
+   *
+   * @param lambda is calculated by multiplying the
```

Comment:
Can you explain what the `lambda` parameter is for instead of how it's calculated?

Reply:
I have updated the definition, kindly have a look and do tell me if it is sufficient.
Thanks.